### PR TITLE
Issue #4713: fix all violations from the third part of idea violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4713

fixed violations remained:
NestedTryStatement
ProtectedField
ProtectedInnerClass
AssignmentToForLoopParameter
ForLoopWithMissingComponent
CheckedExceptionClass
ExtendsConcreteCollection
StaticVariableInitialization
StaticVariableUninitializedUse

### CODE REVIEW

**Comment by Nimfadora** [comment]
@romani while fixing ProtectedField I had to create getter for stream variable in com.puppycrawl.tools.checkstyle.BaseCheckTestSupport as stream in its raw format is used in WriteTagCheckTest#verify

**Comment by romani** [comment]
@Nimfadora , please review violations from TC

File: src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java [hunk 1]
```diff
@@ -220,18 +220,9 @@ public void remove(String checkedFileName) {
...
-            try {
-                oos = new ObjectOutputStream(outputStream);
-                oos.writeObject(object);
-            }
-            finally {
-                flushAndCloseOutStream(oos);
-            }
-
+            // in-memory serialization of Configuration
+            serialize(object, outputStream);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java]: you need to move `oos` out of TRY scope and merge TRYs, so `finally` become part of top TRY.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java [hunk 2]
```diff
@@ -76,19 +76,8 @@
...
-            try {
-                if (!endsWithNewline(randomAccessFile)) {
-                    log(0, MSG_KEY_NO_NEWLINE_EOF, file.getPath());
-                }
-                threw = false;
-            }
-            finally {
-                Closeables.close(randomAccessFile, threw);
-            }
+            readAndCheckFile(file);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java]: you need to move `randomAccessFile` out of TRY scope and merge TRYs, so `finally` become part of top TRY.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java [hunk 3]
```diff
@@ -67,14 +67,7 @@ protected void processFiltered(File file, FileText fileText) {
...
         try {
-            final FileInputStream fileInputStream = new FileInputStream(file);
-            try {
-                // As file is already read, there should not be any exceptions.
-                properties.load(fileInputStream);
-            }
-            finally {
-                fileInputStream.close();
-            }
+            loadProperties(file, properties);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java]: you need to move fileInputStream out of TRY scope and merge TRYs, so finally become part of top TRY.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java [hunk 4]
```diff
@@ -346,6 +341,22 @@ public Tag(String text, int line, SuppressWithNearbyCommentFilter filter) {
             }
         }
 
+        /**
+         * Gets influence from suppress filter influence format param.
+         * @param format influence format to parse
+         * @param influenceFormat raw influence format
+         * @return parsed influence
+         */
+        private int parseInfluence(String format, String influenceFormat) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java]: should be static

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java (hunk 2 above)
Comment (Nimfadora) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java]: Closables.close throws IOException and here the logic is build around threw variable. If we remove it and just put RandomAccessFile creation into try with resources - everything is ok, but than we broke the current logic.

**Comment by rnveach** [comment]
@Nimfadora There is a merge conflict.

**Comment by Nimfadora** [comment]
@romani cobertura fails on my local but I cannot understand how to fix it. [First class](https://nimfadora.github.io/cobertura/com.puppycrawl.tools.checkstyle.PropertyCacheFile.html) and [second class](https://nimfadora.github.io/cobertura/com.puppycrawl.tools.checkstyle.checks.UniquePropertiesCheck.html)

**Comment by romani** [comment]
hmm, sounds familar, .... please try to move content of catch block method and check coverage.
If that fix the problem explain in javadoc of new methods that they exists only due to cobertura bug.

**Comment by rnveach** [comment]
@Nimfadora conversion of try to try-with-resource will almost always lead to cobertura failures.
try-with-resource adds too many extra blocks that we have to cover manually which we don't do right now.
There was a whole discussion in one PR on the different paths that much be taken.

**Comment by rnveach** [comment]
Here is discussion: https://github.com/checkstyle/checkstyle/pull/3963#issuecomment-285795753 (see second block)

**Comment by Nimfadora** [comment]
@romani, @rnveach my first PR variant without fixing comments from @romani passed cobertura normally, as there wasn`t any usage of try-with-resources, I think we should stick with it, what is your opinion?

**Comment by romani** [comment]
> I think we should stick with it, what is your opinion?

please try what i told you to do in my comment for separate method usage .

File: src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java (hunk 1 above)
Comment (Nimfadora) [inline src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java]: Closables.close(outputStream) also fires IOException, so it cannot be resolved in one finally block

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java (hunk 3 above)
Comment (Nimfadora) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java]: fixed

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java (hunk 4 above)
Comment (Nimfadora) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java]: fixed

File: src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java (hunk 1 above)
Comment (Nimfadora) [inline src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java]: Closeables.close(OutputStream) throws IO, leaved it as is

**Pull request merged**